repos/fdbt-site/src/pages/api/managePassengerTypes.ts

```diff
@@ -0,0 +1,216 @@
+                }),
+        }),
+        proofDocuments: yup.string().when('proof', { is: 'Yes', then: yup.string().required(proofSelectionError) }),
+    })
+    .required();
+
+export const formatRequestBody = (req: NextApiRequestWithSession): FilteredRequestBody => {
+    const filteredReqBody: IFilteredRequestBody = {};
+
+    Object.entries(req.body).forEach((entry) => {
+        const key = entry[0];
+        const value = entry[1];
+
+        if (key === 'type') {
+            filteredReqBody[key] = value as string;
+        }
+
+        if (key === 'name') {
+            filteredReqBody[key] = value as string;
+        }
+
+        if (key === 'ageRangeMin' || key === 'ageRangeMax') {
+            const strippedValue = removeAllWhiteSpace(value as string);
+
+            if (strippedValue === '') {
+                return;
+            }
+
+            filteredReqBody[key] = strippedValue;
+
+            return;
+        }
+
+        if (key === 'proofDocuments') {
+            filteredReqBody[key] = !isArray(value) ? [value as string] : (value as string[]);
+
+            return;
+        }
+
+        filteredReqBody[key] = value as string;
+    });
+
+    return filteredReqBody;
+};
+
+export const getErrorIdFromValidityError = (errorPath: string): string => {
+    switch (errorPath) {
+        case 'type':
+            return 'type';
+        case 'name':
+            return 'name';
+        case 'ageRangeMin':
+            return 'age-range-min';
+        case 'ageRangeMax':
+            return 'age-range-max';
+        case 'proofDocuments':
+            return 'membership-card';
+        default:
+            throw new Error(`Could not match the following error with an expected input. Error path: ${errorPath}.`);
+    }
+};
+
+export const getPassengerTypeRedirectLocation = (req: NextApiRequestWithSession, passengerType: string): string => {
+    const { fareType } = getSessionAttribute(req, FARE_TYPE_ATTRIBUTE) as FareType;
+
+    return passengerType === 'schoolPupil' && fareType === 'schoolService' ? '/termTime' : '/defineTimeRestrictions';
+};
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        if (!req.body) {
+            throw new Error('Could not extract the relevant data from the request.');
+        }
+
+        let errors: ErrorInfo[] = [];
+
+        const requestBody = formatRequestBody(req);
+
+        const actualPassengerType = {
+            name: requestBody.name,
+            passengerType: {
+                passengerType: requestBody.type,
+                ageRangeMin: requestBody.ageRangeMin,
+                ageRangeMax: requestBody.ageRangeMax,
+                proofDocuments: requestBody.proofDocuments,
+            },
+        } as SinglePassengerType;
+
+        try {
+            await requestValidationRules.validate(requestBody, { abortEarly: false });
+        } catch (exception) {
+            console.log('ERRORS MAAAAN!!!!!!!');
```

Comment:
Love the passion, but left in accidentally? :stuck_out_tongue_winking_eye:

On this note though, don't we have something that checks for `console.log`s? Should this have been picked up?

Reply:
Removed :) I left it accidentally when testing :+1: